spark/dl/src/main/scala/com/intel/analytics/bigdl/dlframes/DLImageReader.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.dlframes
+
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+import com.intel.analytics.bigdl.transform.vision.image.{BytesToMat, ImageFeature, ImageFrame}
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
+import org.opencv.core.CvType
+
+/**
+ * Definition for image data in a DataFrame
+ */
+object DLImageSchema {
```

Comment:
make it private to  package `dlframes` ?

Reply:
Though it may not be the major use case, but it's possible that user want to manually compose a image column.
The byteSchema is public in Spark Image schema so we probably will keep it stable.

Reply:
Makes sense